Kruskal Algorithm/kruskal.java

```diff
@@ -0,0 +1,233 @@
+	     int b = find(j); 
+	     parent[a] = b; 
+	 } 
+	   
+	 // Finds MST using Kruskal's algorithm 
+	 static void algo(int cost[][],int v) 
+	 { 
+		 for(int i=0;i<v;i++)
+			{
+				for(int j=0;j<v;j++)
+				{
+					if(cost[i][j]==0)
+					{
+						cost[i][j]=999;
+					}
+				}
+					
+
+			}
+	     int mincost = 0; // Cost of min MST. 
+	   
+	   
+		// Initialize sets of disjoint sets. 
+	     for (int i = 0; i < v; i++) 
+	         parent[i] = i; 
+	   
+	     // Include minimum weight edges one by one 
+	     int edge_count = 0; 
+	     while (edge_count < v - 1) 
+	     { 
+	         int min = 999, a = -1, b = -1; 
+	         for (int i = 0; i < v; i++) 
+	         { 
+	             for (int j = 0; j < v; j++)  
+	             { 
+	                 if (find(i) != find(j) && cost[i][j] < min)  
+	                 { 
+	                     min = cost[i][j]; 
+	                     a = i; 
+	                     b = j; 
+	                 } 
+	             } 
+	         } 
+	   
+	         union1(a, b); 
+	         System.out.printf("EDGE %d:(%d, %d) COST:%d \n", 
+	             edge_count++, a+1, b+1, min); 
+	         mincost += min; 
+	     } 
+	     System.out.println(" ");
+	     System.out.println("MINIMUM COST=  "+ mincost); 
+	 }
+}
+public class kruskal {
+
+	public static void main(String[] args) {
+		Scanner s1=new Scanner(System.in);
+		 System.out.println("ENTER NUMBER OF VERTICES(OFFICES)");
+		 int v=s1.nextInt();
+		System.out.println("ENTER NUMBER OF EDGES(ROUTES)");
+		int e=s1.nextInt();
+		graph g=new graph();
+		g.createmat(v, e);
+		
+
+	}
+
+}
+
+/*
+ENTER NUMBER OF VERTICES(OFFICES)
+6
+ENTER NUMBER OF EDGES(ROUTES)
+9
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+1
+ENTER (TO)EDGE FOR EDGE
+5
+ENTER THE COST 
+4
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+1
+ENTER (TO)EDGE FOR EDGE
+4
+ENTER THE COST 
+1
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+5
+ENTER (TO)EDGE FOR EDGE
+4
+ENTER THE COST 
+9
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+4
+ENTER (TO)EDGE FOR EDGE
+3
+ENTER THE COST 
+5
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+4
+ENTER (TO)EDGE FOR EDGE
+2
+ENTER THE COST 
+3
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+2
+ENTER (TO)EDGE FOR EDGE
+3
+ENTER THE COST 
+3
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+3
+ENTER (TO)EDGE FOR EDGE
+6
+ENTER THE COST 
+8
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+6
+ENTER (TO)EDGE FOR EDGE
+2
+ENTER THE COST 
+7
+ 
+ENTER THE VERTICES FOR EDGES
+ENTER (FROM)VERTICE FOR EDGE
+1
+ENTER (TO)EDGE FOR EDGE
+2
+ENTER THE COST 
+2
+  1 2 3 4 5 6  
+_________________________
+1| 0 2 0 1 4 0  
+2| 2 0 3 3 0 7  
+3| 0 3 0 5 0 8  
+4| 1 3 5 0 9 0  
+5| 4 0 0 9 0 0  
+6| 0 7 8 0 0 0  
+ 
+EDGE 0:(1, 4) COST:1 
+EDGE 1:(1, 2) COST:2 
+EDGE 2:(2, 3) COST:3 
+EDGE 3:(1, 5) COST:4 
+EDGE 4:(2, 6) COST:7 
+MINIMUM COST=  17
+*/
```

Comment:
Maybe its a good idea to just put a input sample in this comment :grin:

Reply:
Yes sure.
I will do it!